src/renderer/components/cluster-manager/cluster-view.tsx

```diff
@@ -68,6 +68,11 @@ class NonInjectedClusterView extends React.Component<Dependencies> {
   bindEvents() {
     disposeOnUnmount(this, [
       reaction(() => this.clusterId, async (clusterId) => {
+        // TODO: replace with better handling
+        if (clusterId && !this.props.entityRegistry.getById(clusterId)) {
+          return this.props.navigateToCatalog(); // redirect to catalog when the clusterId does not correspond to an entity
+        }
+
```

Comment:
This code is safe because it is run in the main frame so there should be no timing issues with the catalog not being synced yet.